tests/templates/test_subroutines/test_qsvt.py

```diff
@@ -0,0 +1,329 @@
+                [0, 1],
+                [
+                    qml.RZ(0.1, wires=[0]),
+                    qml.PauliZ(wires=[0]),
+                    qml.RY(0.2, wires=[0]),
+                    qml.adjoint(qml.PauliZ(wires=[0])),
+                    qml.RZ(0.3, wires=[1]),
+                ],
+            ),
+        ],
+    )
+    def test_queuing_ops(self, U_A, lst_projectors, wires, results):
+        """Test that qml.QSVT queues operations in the correct order."""
+        with qml.tape.QuantumTape() as tape:
+            qml.QSVT(U_A, lst_projectors, wires)
+
+        for idx, val in enumerate(tape.expand().operations):
+            assert val.name == results[idx].name
+            assert val.parameters == results[idx].parameters
+
+    def test_queuing_ops_defined_in_circuit(self):
+        """Test that qml.QSVT queues operations correctly when they are called in the qnode."""
+        lst_projectors = [qml.PCPhase(0.2, dim=1, wires=0), qml.PCPhase(0.3, dim=1, wires=0)]
+        wires = [0, 1]
+        results = [
+            qml.PCPhase(0.2, dim=1, wires=[0]),
+            qml.PauliX(wires=[0]),
+            qml.PCPhase(0.3, dim=1, wires=[0]),
+        ]
+
+        with qml.tape.QuantumTape() as tape:
+            qml.QSVT(qml.PauliX(wires=0), lst_projectors, wires)
+
+        for idx, val in enumerate(tape.expand().operations):
+            assert val.name == results[idx].name
+            assert val.parameters == results[idx].parameters
+
+    def qfunc(A):
+        """Used to test queuing in the next test."""
+        return qml.PauliX(wires=0)
+
+    def qfunc2(A):
+        """Used to test queuing in the next test."""
+        return qml.prod(qml.PauliX(wires=0), qml.RZ(A[0][0], wires=0))
+
+    def lst_phis(phis):
+        """Used to test queuing in the next test."""
+        return [qml.PCPhase(i, 2, wires=[0, 1]) for i in phis]
+
+    @pytest.mark.parametrize(
+        ("quantum_function", "phi_func", "A", "phis", "results"),
+        [
+            (
+                qfunc,
+                lst_phis,
+                np.array([[0.1, 0.2], [0.3, 0.4]]),
+                np.array([0.2, 0.3]),
+                [
+                    qml.PCPhase(0.2, dim=2, wires=[0]),
+                    qml.PauliX(wires=[0]),
+                    qml.PCPhase(0.3, dim=2, wires=[0]),
+                ],
+            ),
+            (
+                qfunc2,
+                lst_phis,
+                np.array([[0.1, 0.2], [0.3, 0.4]]),
+                np.array([0.1, 0.2]),
+                [
+                    qml.PCPhase(0.1, dim=2, wires=[0]),
+                    qml.prod(qml.PauliX(wires=0), qml.RZ(0.1, wires=0)),
+                    qml.PCPhase(0.2, dim=2, wires=[0]),
+                ],
+            ),
+        ],
+    )
+    def test_queuing_callables(self, quantum_function, phi_func, A, phis, results):
+        """Test that qml.QSVT queues operations correctly when a function is called inside the qnode"""
```

Comment:
Does this test really performs that? I don't see a qnode defined in the test.

Reply:
Here we're just testing that they get queued in the right order using a tape.

Reply:
Then correct "inside the qnode" to a more appropriate phrase?